server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -585,7 +585,13 @@ private void loadOrExecuteQueryPhase(final ShardSearchRequest request, final Sea
     public void executeQueryPhase(ShardSearchRequest request, CancellableTask task, ActionListener<SearchPhaseResult> listener) {
         assert request.canReturnNullResponseIfMatchNoDocs() == false || request.numberOfShards() > 1
             : "empty responses require more than one shard";
-        final IndexShard shard = getShard(request);
+        final IndexShard shard;
+        try {
+            shard = getShard(request);
+        } catch (RuntimeException e) {
+            listener.onFailure(e);
+            return;
```

Comment:
what is this additional catch fixing? Is it a bug that you observed?

Reply:
Jup concurrent shard deletion can cause a shard not found exception :)

Reply:
should that be a separate change then? It is not a problem introduced by your change, is it?

Reply:
It is introduced here, without this change the transport layer catches the exception and passes it on to the listener.